lib/cylc/network/server.py

```diff
@@ -258,65 +293,140 @@ def __init__(self, schd):
         self.schd = schd
         self.public_priv = None  # update in get_public_priv()
 
-    def get_public_priv(self):
+    def _get_public_priv(self):
         """Return the public privilege level of this suite."""
         if self.schd.config.cfg['cylc']['authentication']['public']:
             return Priv.parse(
                 self.schd.config.cfg['cylc']['authentication']['public'])
         return Priv.parse(glbl_cfg().get(['authentication', 'public']))
 
-    def get_priv_level(self, user):
+    def _get_priv_level(self, user):
         """Return the privilege level for the given user for this suite."""
         if user == getpass.getuser():
             return Priv.CONTROL
         if self.public_priv is None:
             # cannot do this on initialisation as the suite configuration has
             # not yet been parsed
-            self.public_priv = self.get_public_priv()
+            self.public_priv = self._get_public_priv()
         return self.public_priv
 
+    @authorise(Priv.IDENTITY)
+    @ZMQServer.expose
+    def api(self, endpoint=None):
+        """Return information about this API.
+
+        Returns a list of callable endpoints.
+
+        Args:
+            endpoint (str, optional):
+                If specified the documentation for the endpoint
+                will be returned instead.
+
+        Returns:
+            list/str: List of endpoints or string documentation of the
```

Comment:
It's a shame Google docstrings / napoleon / sphinx do not support multiple return types :(. There should be a way to support PEP 484 typing in the generated sphinx docs (probably for later)

Reply:
Yep, I really hate having to specify `optional` for kwargs as well (google docstring standard), why can't it just be inferred from the signature!!!